Automattic/wp-calypso
# Me: Replace security checkup by account recovery

## Summary
This PR replaces the underlying state of `/me/security/checkup` from the old Flux store with the newly-implemented redux store ( see #9208 ).

While everything should work as before, I've moved the update / delete notices from components to the global notices.

## Test plan
* Add / update / delete recovery email should work as expected.
* Add / update / delete recovery phone should work as expected.
* The email field and the phone field should check for validity as expected.
* One shouldn't be able to enter the same recovery email as the primary email.

![demo-checkup](https://cloud.githubusercontent.com/assets/1842898/21215312/5d0d37aa-c2db-11e6-9840-6db2fbef1d27.gif)
